tools/wake-unit/fuzz_test_job_cache.cpp

```diff
@@ -293,6 +298,75 @@ struct Pool {
   }
 };
 
+template <class F>
+bool run_as_init_proc(F f) {
+  // We're going to unshare some namespaces but this can mess with later
+  // things that spawn threads so we want to fork first to isolate
+  int pid_wrapper = fork();
+
+  if (pid_wrapper == -1) {
+    wcl::log::error("run_as_init_proc: fork failure: %s", strerror(errno))();
+    return false;
+  }
+
+  // We wait for the child to return here
+  if (pid_wrapper != 0) {
+    int status = 0;
+    if (waitpid(pid_wrapper, &status, 0) != pid_wrapper) {
+      wcl::log::error("run_as_init_proc: waitpid(): %s", strerror(errno))();
+      return false;
+    }
+
+    // Relay errors up to the top
+    if (WIFEXITED(status) && WEXITSTATUS(status) == 0) {
+      return true;
+    }
+    return false;
+  }
+
+  // Now that we're in an isolated child, we unshare our namespaces
+  if (unshare(CLONE_NEWUSER | CLONE_NEWPID) != 0) {
```

Comment:
optional: I don't think it costs much to add a CLONE_NEWNS here and then a mount of /proc between lines 344-345, it probably makes things easier to reason about if you ever need to debug these in detail

Reply:
wait...I thought I had that code and it was mounted, I think somewhere in here I lost my CLONE_NEWNS and proc mount